source/crud/tests/unified/bulkWrite-updateOne-let.yml

```diff
@@ -0,0 +1,104 @@
+description: "BulkWrite updateOne-let"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1, x: 20 }
+      - { _id: 2, x: 21 }
+
+tests:
+  - description: "BulkWrite updateOne with let option"
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            - updateOne:
+                filter: &filter
+                  _id:
+                    $gt: "$$id"
+                update: &update
+                  $inc:
+                    x: 1
+                let: &let
+                  id: 1
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                update: *collection0Name
+                updates:
+                  - q: *filter
+                    u: *update
+                    let: *let
+                    multi:
+                      $$unsetOrMatches: false
+                    upsert:
+                      $$unsetOrMatches: false
+    outcome:
+      -
+        collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - { _id: 1, x: 20 }
+          - { _id: 2, x: 22 }
+
+  - description: "Bulk Write updateOne with let option unsupported
+  (server-side error)"
+    runOnRequirements:
+      - minServerVersion: "3.6.0"
+        maxServerVersion: "4.9"
+    operations:
+      - object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            - updateOne:
+                filter: &filter
```

Comment:
We're redefining "filter" and other labels here, is that allowed?

Reply:
Based on the output of `js-yaml` it appears to be allowed.

Reply:
According to Google, some parsers have had bugs where they don't allow this, but it's not prohibited in the YAML spec and we do make use of this in existing spec tests. When an alias is resolved, it will evaluate the the most recently (re)defined anchor.

That said, there's no reason to redefine `&filter` here since we're using the same value as before. This should be replaced with an alias to `*filter`.

Reply:
Got rid of this and other instances where I could have reused the same anchor.